fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePrimaryKeySegmentIndex.java

```diff
@@ -76,15 +78,21 @@ public LucenePrimaryKeySegmentIndex(@Nonnull FDBDirectory directory, @Nonnull Su
      */
     @VisibleForTesting
     public List<List<Object>> readAllEntries() {
+        AtomicReference<List<List<Object>>> list = new AtomicReference<>();
+        directory.getAgilityContext().accept(aContext -> readAllEntries(aContext, list));
```

Comment:
Why use `accept` with the `AtomicReference`, rather than just calling `apply`?
I think you would have to move the `asyncToSync` up to this method.

Reply:
I've tried that first, and it broke.